lib/src/license_detection/tokenizer.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:collection';
+import 'dart:convert';
+
+class Token {
```

Comment:
Unless we think someone should subclass `Token` which seems like a bad idea... maybe we can do:
```dart
@sealed
class Token {
....
```

The annotation `@sealed` comes from `package:meta` which is fine to use...